File: Day6/MajorityElementII.py
# Link:- https://www.geeksforgeeks.org/batch/gfg-160-problems/track/arrays-gfg-160/problem/majority-vote

from collections import Counter
def MajorityElement(n,arr):
    new_freq_arr=[]
    freq= Counter(arr)
    for i,j in sorted(freq.items()):
        if(j>(n/3)):
            new_freq_arr.append(i)
    return new_freq_arr


n = int(input())
arr=list(map(int,input().split()))
new_arr = MajorityElement(n,arr)
for i in new_arr:
    print(i)

File: Day6/MajorityElementII.cpp
// Link:- https://www.geeksforgeeks.org/batch/gfg-160-problems/track/arrays-gfg-160/problem/majority-vote
#include<bits/stdc++.h>
using namespace std;

vector<int> new_majority(int n, vector<int>& arr){
    
    map<int,int> freq;
    for(auto &ar:arr){
        freq[ar]++;
    }
    
    vector<int> new_arr;
    for(auto ne_ar:freq){
        if(ne_ar.second>(n/3)){
            new_arr.push_back(ne_ar.first);
        }
    }
    
    return new_arr;
    
    
    
}

int main(){
    int n;
    cin>>n;
    int i,m;
    vector<int> arr;
    for(i=0;i<n;i++){
        cin>>m;
        arr.push_back(m);
        
    }
    
    vector<int> new_freq_arr;
    
    new_freq_arr = new_majority(n,arr);
    
    for(auto n_f_a:new_freq_arr){
        cout<<n_f_a<<endl;
    }
    
    
    
}